File: app/src/main/java/ntktest/example/com/ntktest/presentation/FeedAdapter.java
package ntktest.example.com.ntktest.presentation;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import ntktest.example.com.ntktest.R;
import ntktest.example.com.ntktest.data.RssItem;

public class FeedAdapter extends RecyclerView.Adapter<FeedAdapter.ViewHolder> {
    public interface OnItemClickListener {
        void onClick(int position);
    }

    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        @BindView(R.id.title)
        public TextView title;

        @BindView(R.id.date)
        public TextView date;

        @BindView(R.id.source)
        public TextView source;

        public ViewHolder(@NonNull View item) {
            super(item);
            ButterKnife.bind(this, item);
            item.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            if (clickListener == null || getAdapterPosition() == RecyclerView.NO_POSITION) {
                return;
            }
            clickListener.onClick(getAdapterPosition());
        }
    }

    @NonNull
    private List<RssItem> data = new ArrayList<>();

    @Nullable
    private OnItemClickListener clickListener;

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater
                .from(parent.getContext())
                .inflate(R.layout.item_feed, parent, false);

        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        RssItem item = data.get(position);

        holder.title.setText(item.getTitle());
        holder.date.setText(item.getPubDate());
        holder.source.setText("Meduza");
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public void setOnItemClickListener(@Nullable OnItemClickListener listener) {
        clickListener = listener;
    }

    @NonNull
    public List<RssItem> getData() {
        return data;
    }

    public void setData(@NonNull List<RssItem> data) {
        this.data = data;
    }
}

File: app/src/main/java/com/example/rssreader/presentation/feed/FeedAdapter.kt
package com.example.rssreader.presentation.feed

import android.view.View
import android.view.ViewGroup
import com.example.rssreader.R
import com.example.rssreader.domain.entity.RssItem
import com.example.rssreader.presentation.BaseAdapter
import com.example.rssreader.presentation.BaseViewHolder
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_feed.*

class FeedAdapter : BaseAdapter<FeedAdapter.ViewHolder>() {

    inner class ViewHolder(override val containerView: View, clickListener: (Int) -> Unit)
        : BaseViewHolder(containerView, clickListener), LayoutContainer

    var data: List<RssItem> = emptyList()
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = inflate(R.layout.item_feed, parent)
        return ViewHolder(view, onItemClickListener)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = data[position]

        holder.title.text = item.title
        holder.date.text = item.pubDate
        holder.source.text = "Meduza"
    }

    override fun getItemCount(): Int = data.size
}